From a27dcd387d8c3c1f1e150dcdd3c8aa1872ad14b5 Mon Sep 17 00:00:00 2001
From: Mike Perry <mikeperry-git@fscked.org>
Date: Tue, 20 Dec 2011 21:02:49 -0800
Subject: [PATCH 10/16] Provide client values only to CSS Media Queries

Also disable a bunch of Mozilla extensions that smell like they are
fingerprintable.

This is done to address
https://www.torproject.org/projects/torbrowser/design/#fingerprinting-linkability
---
 layout/style/nsMediaFeatures.cpp |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/layout/style/nsMediaFeatures.cpp b/layout/style/nsMediaFeatures.cpp
index 6eca06e..c68f191 100644
--- a/layout/style/nsMediaFeatures.cpp
+++ b/layout/style/nsMediaFeatures.cpp
@@ -383,14 +383,14 @@ nsMediaFeatures::features[] = {
         nsMediaFeature::eMinMaxAllowed,
         nsMediaFeature::eLength,
         { nsnull },
-        GetDeviceWidth
+        GetWidth
     },
     {
         &nsGkAtoms::deviceHeight,
         nsMediaFeature::eMinMaxAllowed,
         nsMediaFeature::eLength,
         { nsnull },
-        GetDeviceHeight
+        GetHeight
     },
     {
         &nsGkAtoms::orientation,
@@ -411,7 +411,7 @@ nsMediaFeatures::features[] = {
         nsMediaFeature::eMinMaxAllowed,
         nsMediaFeature::eIntRatio,
         { nsnull },
-        GetDeviceAspectRatio
+        GetAspectRatio
     },
     {
         &nsGkAtoms::color,
@@ -457,6 +457,7 @@ nsMediaFeatures::features[] = {
     },
 
     // Mozilla extensions
+/*
     {
         &nsGkAtoms::_moz_device_pixel_ratio,
         nsMediaFeature::eMinMaxAllowed,
@@ -469,7 +470,7 @@ nsMediaFeatures::features[] = {
         nsMediaFeature::eMinMaxNotAllowed,
         nsMediaFeature::eEnumerated,
         { kOrientationKeywords },
-        GetDeviceOrientation
+        GetOrientation
     },
     {
         &nsGkAtoms::_moz_is_resource_document,
@@ -590,6 +591,7 @@ nsMediaFeatures::features[] = {
         { nsnull },
         GetWindowsTheme
     },
+*/
     // Null-mName terminator:
     {
         nsnull,
-- 
1.7.5.4

